cluster/gce/windows/k8s-node-setup.psm1

```diff
@@ -1525,349 +1525,423 @@ function Start_Containerd {
   Log-Output "Starting containerd service"
   Start-Service containerd
 }
-
-# TODO(pjh): move the Stackdriver logging agent code below into a separate
+# TODO(pjh): move the logging agent code below into a separate
 # module; it was put here temporarily to avoid disrupting the file layout in
 # the K8s release machinery.
-$STACKDRIVER_VERSION = 'v1-11'
-$STACKDRIVER_ROOT = 'C:\Program Files (x86)\Stackdriver'
+$LOGGINGAGENT_VERSION = 'v1-11'
+$LOGGINGAGENT_INSTALLER_DIR = 'C:\flb-installers'
+$LOGGINGAGENT_ROOT = 'C:\fluent-bit'
+$LOGGINGAGENT_SERVICE = 'fluent-bit'
+$LOGGINGAGENT_CMDLINE = '*fluent-bit.exe*'
+
+$LOGGINGEXPORTER_ROOT = 'C:\flb-exporter'
+$LOGGINGEXPORTER_SERVICE = 'flb-exporter'
+$LOGGINGEXPORTER_BINARY = 'C:\flb-exporter\flb-exporter.exe'
+$LOGGINGEXPORTER_CMDLINE = '*flb-exporter.exe*'
+
+# Restart Logging agent or starts it if it is not currently running
+function Restart-LoggingAgent {
+   Restart-LogService $LOGGINGEXPORTER_SERVICE $LOGGINGEXPORTER_CMDLINE
+   Restart-LogService $LOGGINGAGENT_SERVICE $LOGGINGAGENT_CMDLINE
+}
 
-# Restarts the Stackdriver logging agent, or starts it if it is not currently
-# running. A standard `Restart-Service StackdriverLogging` may fail because
-# StackdriverLogging sometimes is unstoppable, so this function works around it
+# Restarts the service, or starts it if it is not currently
+# running. A standard `Restart-Service` may fail because
+# the process is sometimes unstoppable, so this function works around it
```

Comment:
Do you know if we expect this behavior to be better or the same now with fluent-bit?

Reply:
I had seen it occur in one of the services (fluent-bit, flb-exporter) when testing manually.